torchgeo/trainers/change.py

```diff
@@ -311,7 +304,7 @@ def _shared_step(self, batch: Any, batch_idx: int, stage: str) -> Tensor:
                     case 'binary' | 'multilabel':
                         batch['prediction'] = (y_hat.sigmoid() >= 0.5).long()
                     case 'multiclass':
-                        batch['prediction'] = y_hat.argmax(dim=1)
+                        batch['prediction'] = y_hat.argmax(dim=1, keepdim=True)
```

Comment:
Could also change all multiclass CD datasets to assume prediction has no channel dimension. But then prediction and mask will not have consistent dimensions.